components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/AdminForcedPasswordResetHandler.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations und
+ */
+
+package org.wso2.carbon.identity.recovery.handler;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.common.model.User;
+import org.wso2.carbon.identity.core.bean.context.MessageContext;
+import org.wso2.carbon.identity.event.IdentityEventConstants;
+import org.wso2.carbon.identity.event.IdentityEventException;
+import org.wso2.carbon.identity.event.event.Event;
+import org.wso2.carbon.identity.recovery.IdentityRecoveryConstants;
+import org.wso2.carbon.identity.recovery.RecoveryScenarios;
+import org.wso2.carbon.identity.recovery.RecoverySteps;
+import org.wso2.carbon.identity.recovery.util.Utils;
+import org.wso2.carbon.user.core.UserStoreManager;
+
+import java.util.Map;
+
+public class AdminForcedPasswordResetHandler extends UserEmailVerificationHandler {
+
+    private static final Log log = LogFactory.getLog(AdminForcedPasswordResetHandler.class);
+
+    @Override
+    public void handleEvent(Event event) throws IdentityEventException {
+        if (IdentityEventConstants.Event.PRE_SET_USER_CLAIMS.equals(event.getEventName())) {
+            Map<String, Object> eventProperties = event.getEventProperties();
+            UserStoreManager userStoreManager = (UserStoreManager) eventProperties.get(IdentityEventConstants.EventProperty.USER_STORE_MANAGER);
+            User user = getUser(eventProperties, userStoreManager);
+
+            Map<String, String> claims = (Map<String, String>) eventProperties.get(IdentityEventConstants.EventProperty
+                    .USER_CLAIMS);
+
+            boolean adminPassResetOffline = Boolean.parseBoolean(Utils.getConnectorConfig(
```

Comment:
Where are we using adminPassResetOffline, and adminPassResetOTP congis ?
Are we not storing a code for them ?
If adminPassResetRecoveryLink is false